pkg/host/host_linux.go

```diff
@@ -316,14 +316,57 @@ func isSupportedOpenAt(c *prog.Syscall) (bool, string) {
 	if !ok || len(fname) == 0 || fname[0] != '/' {
 		return true, ""
 	}
+
+	// Test 1: try to extract the correct flags
+	constFlags, ok := c.Args[2].(*prog.ConstType)
+	if ok {
+		// flag exists and is const
+		fd, err := syscall.Open(fname, int(constFlags.Val), 0)
```

Comment:
It looks like the same piece of code is duplicated 4 times and there is an easy to remove this duplication, something along the lines of:

```
	modes := []int{syscall.O_RDONLY, syscall.O_WRONLY, syscall.O_RDWR}
	if mode, ok := c.Args[2].(*prog.ConstType); ok {
		modes = []int{int(mode.Val)}
	}
	for _, mode := range modes {
		...
	}
```

Reply:
See new updates, I believe I have addressed this concern?